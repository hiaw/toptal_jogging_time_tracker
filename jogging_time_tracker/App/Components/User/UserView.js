/* @flow*/
import React from 'react'
import { Image, View, Button } from 'react-native'
import { Field } from 'redux-form'

import {
  required,
  email,
  minLength,
  maxLength,
} from '../../Helper/Validators.js'

import BottomButtons from './BottomButtons.js'
import FormFieldText from '../Common/FormFieldText.js'
import FormFieldSelect from '../Common/FormFieldSelect.js'
import type { Props } from '../TimeRow/TimeLogView.js'

const styles = {
  container: {
    marginTop: 65,
  },
}

const options = [
  {
    label: 'Admin',
    value: 'admin',
  },
  {
    label: 'Manager',
    value: 'manager',
  },
  {
    label: 'User',
    value: '',
  },
]

const UserView = (props: Props) => {
  const {
    role,
    valid,
    newEntry,
    editing,
    handleSubmit,
    onSubmit,
    pickImage,
    showTimelogs,
  } = props

  let showTimelogsButton = null
  if (role === 'admin') {
    showTimelogsButton = <Button onPress={showTimelogs} title="Show Timelogs" />
  }

  let selectRole = (
    <Field
      component={FormFieldSelect}
      name="role"
      title="ROLE"
      editable={editing || newEntry}
      options={options}
    />
  )
  if (role === 'user') {
    selectRole = null
  }

  return (
    <View style={styles.container}>
      {showTimelogsButton}
      <Field
        autoCapitalize="none"
        component={FormFieldText}
        name="email"
        title="EMAIL"
        editable={editing || newEntry}
        validate={[required, email, minLength(2), maxLength(30)]}
      />
      <Field
        component={FormFieldText}
        name="password"
        title="PASSWORD"
        secureTextEntry
        editable={editing || newEntry}
        validate={[minLength(2), maxLength(30)]}
      />
      <Button title="upload image" onPress={pickImage} />
      {selectRole}
      <BottomButtons {...props} />
    </View>
  )
}

export default UserView
