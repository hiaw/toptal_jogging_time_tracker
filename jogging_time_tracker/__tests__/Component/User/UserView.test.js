import 'react-native'
import React from 'react'
import { reduxForm } from 'redux-form'
import { createStore } from 'redux'
import { Provider } from 'react-redux'

import UserView from '../../../App/Components/User/UserView.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const noop = () => {}

const store = createStore(() => ({}))

const Decorated = reduxForm({ form: 'user_form' })(UserView)

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Decorated
          buttonText="Submit"
          valid={true}
          newEntry={true}
          editing={true}
          alterEditing={noop}
          handleSubmit={noop}
          onSubmit={noop}
          deleteTimeLog={noop}
          cancelEditing={noop}
        />
      </Provider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders invalid', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Decorated
          buttonText="Submit"
          valid={false}
          newEntry={true}
          editing={true}
          alterEditing={noop}
          handleSubmit={noop}
          onSubmit={noop}
          deleteTimeLog={noop}
          cancelEditing={noop}
        />
      </Provider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders not editing', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Decorated
          buttonText="Submit"
          valid={false}
          newEntry={false}
          editing={false}
          alterEditing={noop}
          handleSubmit={noop}
          onSubmit={noop}
          deleteTimeLog={noop}
          cancelEditing={noop}
        />
      </Provider>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
