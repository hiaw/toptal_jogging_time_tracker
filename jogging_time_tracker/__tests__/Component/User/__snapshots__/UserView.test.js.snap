// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders admin 1`] = `
<View
  style={
    Object {
      "marginTop": 65,
    }
  }
>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Show Timelogs
      </Text>
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        EMAIL
        :
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          editable={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Tap to write"
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          testID="textQuestionInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        PASSWORD
        :
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          editable={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Tap to write"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          testID="textQuestionInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={undefined}
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Admin",
            "textColor": undefined,
            "value": "admin",
          },
          Object {
            "label": "Manager",
            "textColor": undefined,
            "value": "manager",
          },
          Object {
            "label": "User",
            "textColor": undefined,
            "value": "",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={2}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "marginTop": 65,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        EMAIL
        :
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Tap to write"
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          testID="textQuestionInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        PASSWORD
        :
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Tap to write"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          testID="textQuestionInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={undefined}
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Admin",
            "textColor": undefined,
            "value": "admin",
          },
          Object {
            "label": "Manager",
            "textColor": undefined,
            "value": "manager",
          },
          Object {
            "label": "User",
            "textColor": undefined,
            "value": "",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={2}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders invalid 1`] = `
<View
  style={
    Object {
      "marginTop": 65,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        EMAIL
        :
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Tap to write"
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          testID="textQuestionInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        PASSWORD
        :
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          editable={true}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Tap to write"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          testID="textQuestionInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={undefined}
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Admin",
            "textColor": undefined,
            "value": "admin",
          },
          Object {
            "label": "Manager",
            "textColor": undefined,
            "value": "manager",
          },
          Object {
            "label": "User",
            "textColor": undefined,
            "value": "",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={2}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Cancel
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders not editing 1`] = `
<View
  style={
    Object {
      "marginTop": 65,
    }
  }
>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        EMAIL
        :
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          autoCapitalize="none"
          editable={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Tap to write"
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          testID="textQuestionInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {},
          undefined,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {},
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginBottom": 1,
                "marginLeft": 20,
                "marginRight": 20,
                "marginTop": 15,
              },
              undefined,
              undefined,
            ],
          ]
        }
      >
        PASSWORD
        :
         
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#bdc6cf",
              "borderBottomWidth": 1,
              "marginLeft": 20,
              "marginRight": 20,
            },
            undefined,
          ]
        }
      >
        <TextInput
          editable={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Tap to write"
          secureTextEntry={true}
          style={
            Array [
              Object {
                "color": "#86939e",
                "fontSize": 17.5,
                "minHeight": 36,
                "width": 750,
              },
              undefined,
            ]
          }
          testID="textQuestionInput"
          underlineColorAndroid="transparent"
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={undefined}
  >
    <RCTPicker
      items={
        Array [
          Object {
            "label": "Admin",
            "textColor": undefined,
            "value": "admin",
          },
          Object {
            "label": "Manager",
            "textColor": undefined,
            "value": "manager",
          },
          Object {
            "label": "User",
            "textColor": undefined,
            "value": "",
          },
        ]
      }
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      selectedIndex={2}
      style={
        Array [
          Object {
            "height": 216,
          },
          undefined,
        ]
      }
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
